R/randomForest_tidiers.R

```diff
@@ -178,6 +178,113 @@ randomForestMulti <- function(data, formula, na.action = na.omit, ...) {
   ret
 }
 
+rangerCore <- function(data, formula, na.action = na.omit,
+                       importance_mode = "permutation",
+                       model_type = "regression", ...) {
+  target_col <- all.vars(formula)[[1]]
+  original_val <- data[[target_col]]
+  original_colnames <- colnames(data)
+
+  target_col_index <- which(colnames(data) == target_col)
+
+  # randomForest must take clean names
+  data <- janitor::clean_names(data)
+  updated_colnames <- colnames(data)
+  # this will be used to get original colmn names later
+  names(updated_colnames) <- original_colnames
+  # get target col as clean name
+  target_col <- colnames(data)[target_col_index]
+
+  if(model_type %in% c("classification_binary", "classification_multi")){
+    data[[target_col]] <- as.factor(data[[target_col]])
+  }
+
+  if(!is.logical(original_val) &&
+     length(levels(data[[target_col]] )) != 2 && model_type == "classification_binary"){
+    stop("There should be 2 unique values for binary classification.")
+  }
+
+  if("." %in% all.vars(lazyeval::f_rhs(formula))){
+    original_target_col <- all.vars(formula)[1]
+    vars <- original_colnames[original_colnames != original_target_col]
+    newvars <- updated_colnames[vars]
+  } else {
+    vars <- all.vars(formula)[-1]
+    newvars <- updated_colnames[vars]
+  }
+  formula <- as.formula(paste0(target_col, " ~ ", paste0(newvars, collapse = " + ")))
+
+  # ranger::ranger can't build model when there are NA values in data
+  names(newvars) <- NULL
+  is_na_atrow <- data %>% dplyr::select(target_col, newvars) %>% is.na() %>% apply(1, any)
```

Comment:
Some explanation about this variable might help me follow the logic.

Reply:
The ranger :: raner will generate an error if it contains NA for both explanatory variables and target variables.
Therefore, you need to exclude rows that have NA values ​​in any of those columns.
Since it is necessary to save the information of the column excluded by NA and use it later, the true / false value of which row has NA value once is judged.